### ISSUE TITLE
Cannot force page breaks after or before large table rows

### ISSUE THREAD

**Issue by scottcamp** [wkhtmltopdf/wkhtmltopdf#2997]
Using wkhtmltopdf 0.12.3-dev-79ff51e (with patched qt), I am generating a PDF with a lot of tables. Some of the rows in some of the tables span multiple pages.

Specifying a page break before OR after a row seems to be ignored by wkhtmltopdf. 

I don't have floating parent items and no overflow as other posts suggest may be an issue.

Below is the CSS as seen from the client before creating the PDF for the offending element (tr):

`
element.style {
    page-break-after: always;
}
.tabulated-items-row {
    border-bottom: 1px solid #d0d0d0;
}

tr {
    page-break-inside: avoid;
}
*, *:before, *:after {
    box-sizing: inherit;
}
user agent stylesheet
tr {
    display: table-row;
    vertical-align: inherit;
    border-color: inherit;
}
Inherited from table
table {
    border-collapse: collapse;
    border-spacing: 0;
}
user agent stylesheet
table {
    display: table;
    border-collapse: separate;
    border-spacing: 2px;
    border-color: grey;
}
Inherited from html
@media only screen and (min-width: 0)
html {
    font-size: 14px;
}
html {
    line-height: 1.5;
    font-family: "Roboto",sans-serif;
    font-weight: normal;
    color: rgba(0,0,0,0.87);
}
html {
    font-family: sans-serif;
    -ms-text-size-adjust: 100%;
    -webkit-text-size-adjust: 100%;
}
Pseudo ::before element
*, *:before, *:after {
    box-sizing: inherit;
}
Pseudo ::after element
*, *:before, *:after {
    box-sizing: inherit;
}`

I'd like to understand if this is a limitation of wkhtmltopdf and if others have come across a similar problem?

**Comment by darthnorman** [comment]
see http://caniuse.com/#search=page-break
you can use page-break-inside: avoid; to avoid breaks inside of an element (so the page breaks before or after that element). If you want to force a break before or after a specific element, you have to use page-break-before/after: always; since avoid is not supported for those properties.

**Comment by engineercamp** [comment]
@darthnorman my problem is that "page-break-before/after: always" has no effect.

**Comment by joshjacobson** [comment]
@engineercamp "page-break-before/after: always" should be "page-break-before: always" or "page-break-after: always"

**Comment by engineercamp** [comment]
@joshjacobson yes, I know (see the original css I posted). Neither are getting "actioned" by wkhtmltopdf.

**Comment by cindyliu-yb** [comment]
Hello, same here! any update?

**Comment by JohnnyBDude** [comment]
Hello I have the same problem. Have you guys had any luck getting around that?
Any hint would be greatly appriciated. Thanks

**Comment by andkorsh** [comment]
@JohnnyBDude @cindyliu-yb Take a look at my pull request - it should solve your problem https://github.com/wkhtmltopdf/qt/pull/36

**Comment by corroded** [comment]
Since the PR is not merged yet, do you just use your own fork @andkorsh ? We're also stuck in `0.12.3` right now. Not sure if upgrading would actually help given the PR is still unmerged 😞

**Comment by andkorsh** [comment]
Hi @corroded, you can just use my fork for now https://github.com/andkorsh/wkhtmltopdf
Should be linked with correct version of QT, please let me know if it doesn't work for you.

Here's how you can build the code https://github.com/wkhtmltopdf/wkhtmltopdf/blob/master/INSTALL.md

**Comment by corroded** [comment]
Thanks @andkorsh! Will check it out some time! 👍

**Comment by bugallos** [comment]
Hello!
@andkorsh , can i build the code on CentOS 6.4? I'm not linux expert and don't want to break server in my work. It's a red-hat based distro.  I don't understand what "you will need to use the posix-local target" means, as you wrote in description.
Thanks!

**Comment by andkorsh** [comment]
This is not my description, I just did the PR.
No idea on red head stuff sorry

30 нояб. 2017 г. 5:42 PM пользователь "bugallos" <notifications@github.com>
написал:

Hello!
@andkorsh <https://github.com/andkorsh> , can i build the code on CentOS
6.4? I'm not linux expert and don't want to break server in my work. It's a
red-hat based distro. I don't understand what "you will need to use the
posix-local target" means, as you wrote in description.
Thanks!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2997#issuecomment-348207561>,
or mute the thread

**Comment by saddamhussain734** [comment]
adding div with page break attribute in td solved it.
